lib/ui/viewer/search/location_collection_page.dart

```diff
@@ -0,0 +1,86 @@
+import 'package:flutter/material.dart';
+import 'package:photos/core/event_bus.dart';
+import 'package:photos/events/files_updated_event.dart';
+import 'package:photos/models/file_load_result.dart';
+import 'package:photos/models/gallery_type.dart';
+import 'package:photos/models/search/location_search_result.dart';
+import 'package:photos/models/selected_files.dart';
+import 'package:photos/ui/viewer/gallery/gallery.dart';
+import 'package:photos/ui/viewer/gallery/gallery_app_bar_widget.dart';
+import 'package:photos/ui/viewer/gallery/gallery_overlay_widget.dart';
+
+class LocationCollectionPage extends StatelessWidget {
+  final LocationSearchResult locationAndFiles;
+  final String tagPrefix;
+  final GalleryType appBarType;
+  final GalleryType overlayType;
+  final _selectedFiles = SelectedFiles();
+
+  LocationCollectionPage({
+    this.locationAndFiles,
+    this.tagPrefix = "location_search",
+    this.appBarType = GalleryType.searchResults,
+    this.overlayType = GalleryType.searchResults,
+    Key key,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final gallery = Gallery(
+      asyncLoader: (creationStartTime, creationEndTime, {limit, asc}) {
+        final result = locationAndFiles.files
+            .where(
+              (file) =>
+                  file.creationTime >= creationStartTime &&
+                  file.creationTime <= creationEndTime,
+            )
+            .toList();
+        return Future.value(
+          FileLoadResult(result, result.length < locationAndFiles.files.length),
+        );
+      },
+      reloadEvent: Bus.instance.on<FilesUpdatedEvent>().where(
+            (event) =>
+                event.updatedFiles.firstWhere(
+                  (element) => element.uploadedFileID != null,
+                  orElse: () => null,
+                ) !=
+                null,
+          ),
+      forceReloadEvents: [
+        Bus.instance.on<FilesUpdatedEvent>().where(
+              (event) =>
+                  event.updatedFiles.firstWhere(
+                    (element) => element.uploadedFileID != null,
+                    orElse: () => null,
+                  ) !=
+                  null,
+            ),
+      ],
+      tagPrefix: tagPrefix,
+      selectedFiles: _selectedFiles,
+      initialFiles: [locationAndFiles.files[0]],
+      footer: const SizedBox(height: 120),
```

Comment:
Why do we need to pass this?

Reply:
The footer is for extra space on the bottom of the collection when scrolled down completely. This is needed when the selection overlay is visible.